contracts/strategies/mstable/IdleMStableStrategy.sol

```diff
@@ -0,0 +1,299 @@
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+
+/// @author IdleHusbandry.
+/// @title IdleMStableStrategy
+/// @notice IIdleCDOStrategy to deploy funds in Idle Finance
+/// @dev This contract should not have any funds at the end of each tx.
+contract IdleMStableStrategy is Initializable, OwnableUpgradeable, ERC20Upgradeable, ReentrancyGuardUpgradeable, IIdleCDOStrategy {
+    using SafeERC20Upgradeable for IERC20Detailed;
+
+    /// @notice underlying token address (eg mUSD)
+    address public override token;
+
+    /// @notice address of the strategy used, in this case imUSD
+    address public override strategyToken;
+
+    /// @notice decimals of the underlying asset
+    uint256 public override tokenDecimals;
+
+    /// @notice one underlying token
+    uint256 public override oneToken;
+
+    /// @notice idleToken contract
+    ISavingsContractV2 public imUSD;
+
+    /// @notice underlying ERC20 token contract
+    IERC20Detailed public underlyingToken;
+
+    /* ------------Extra declarations ---------------- */
+    /// @notice address of the governance token. (Here META)
+    address public govToken;
+
+    /// @notice vault
+    IVault public vault;
+
+    /// @notice address of the IdleCDO
+    address public idleCDO;
+
+    /// @notice uniswap router path that should be used to swap the tokens
+    address[] public uniswapRouterPath;
+
+    /// @notice interface derived from uniswap router
+    IUniswapV2Router02 public uniswapV2Router02;
+
+    /// @notice amount last indexed for calculating APR
+    uint256 public lastIndexAmount;
+
+    /// @notice time when last deposit/redeem was made, used for calculating the APR
+    uint256 public lastIndexedTime;
+
+    /// @notice one year, used to calculate the APR
+    uint256 constant YEAR = 365 days;
+
+    constructor() {
+        token = address(1);
+    }
+
+    /// @notice Can be called only once
+    /// @dev Initialize the upgradable contract
+    /// @param _strategyToken address of the strategy token. Here imUSD
+    /// @param _underlyingToken address of the token deposited. here mUSD
+    /// @param _vault address of the of the vault
+    /// @param _idleCDO address of the idleCDO contract
+    /// @param _uniswapV2Router02 address of the uniswap router
+    /// @param _routerPath path to swap the gov tokens
+    function initialize(
+        address _strategyToken,
+        address _underlyingToken,
+        address _vault,
+        address _idleCDO,
+        address _uniswapV2Router02,
+        address[] calldata _routerPath,
+        address _owner
+    ) public initializer {
+        OwnableUpgradeable.__Ownable_init();
+        ReentrancyGuardUpgradeable.__ReentrancyGuard_init();
+
```

Comment:
let's add a require for `token == address(0)` so to avoid initialization of the logic contract (given the `token = address(1);` in the constructor)